File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/security/jwt/JwtIssuerProperties.java
package br.com.deltaglobalbank.identity.infrastructure.security.jwt;

import java.time.Duration;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.bind.DefaultValue;

@ConfigurationProperties(prefix = "identity.jwt")
public record JwtIssuerProperties(
    @DefaultValue("identity-service") String issuer,
    @DefaultValue("internal") String audience,
    @DefaultValue("PT15M") Duration accessTokenTtl,
    @DefaultValue("P7D") Duration refreshTokenTtl
) {
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/security/jwt/JwtIssuerProperties.kt
package br.com.deltaglobalbank.identity.infrastructure.security.jwt

import org.springframework.boot.context.properties.ConfigurationProperties
import java.time.Duration

@ConfigurationProperties(prefix = "identity.jwt")
data class JwtIssuerProperties(
    val issuer: String = "identity-service",
    val audience: String = "internal",
    val accessTokenTtl: Duration = Duration.ofMinutes(15),
    val refreshTokenTtl: Duration = Duration.ofDays(7)
)
